### PR TITLE
amp-ad-exit: use HostService if exists.

### PR DESCRIPTION
**Pull request by lannka**
this is a draft to demonstrate how extension can hook up with `HostService`.
please take a look.

will do some manual testing on real environment before merging.

for #14365

### CODE REVIEW

File: extensions/amp-ad-exit/0.1/amp-ad-exit.js [hunk 1]
```diff
@@ -106,7 +108,21 @@ export class AmpAdExit extends AMP.BaseElement {
...
     }
-    openWindowDialog(this.win, substituteVariables(target.finalUrl), '_blank');
+    const finalUrl = substituteVariables(target.finalUrl);
+    if (HostServices.isAvailable(this.getAmpDoc())) {
+      HostServices.exitForDoc(this.getAmpDoc())
+          .then(exitService => exitService.openUrl(finalUrl))
+          .catch(errorCode => {
+            // TODO: reporting on errors
+            dev().fine(TAG, 'HostServiceError: ' + errorCode);
+            // fallback on browser API on environment miss match
```
Comment (clawr) [inline extensions/amp-ad-exit/0.1/amp-ad-exit.js]: what about other errors?

Comment (lannka) [reply extensions/amp-ad-exit/0.1/amp-ad-exit.js]: report errors, as a TODO

File: src/inabox/host-services.js [hunk 2]
```diff
@@ -25,8 +26,27 @@ const ServiceNames = {
   EXIT: 'host-exit',
 };
 
+export const HostServiceError = {
+  MISS_MATCH: 1, // The HostService impl mis-matches the environment,
```
Comment (jeffkaufman) [inline src/inabox/host-services.js]: This should probably be spelled `MISMATCH` since "mismatch" is a word

File: src/inabox/host-services.js [hunk 3]
```diff
@@ -25,8 +26,27 @@ const ServiceNames = {
...
+  // extension should consider to fallback to
+  // its default behavior on regular web.
+  NOT_SUPPORTED: 2, // The particular feature is not supported
+  // by the current implementation of HostServices,
+  // it's up to extension to decide what's the best
+  // fallback behavior.
+};
```
Comment (jeffkaufman) [inline src/inabox/host-services.js]: What about:

```
export const HostServiceError = {
  // The host service doesn't match its environment.  For example, a SafeFrame
  // host service when run in something that isn't a SafeFrame.  The
  // implementation should consider falling back to its default implementation
  // on the regular web.
  MISMATCH: 1,
  // The host service is correct for its environment, but not able to function.
  // For example, a SafeFrame host service running inside a SafeFrame
  // implementation that is incomplete or out of date.
  NOT_SUPPORTED: 2,
};
```

File: src/inabox/host-services.js (hunk 2 above)
Comment (lannka) [reply src/inabox/host-services.js]: done (after merging with master)

File: src/inabox/host-services.js (hunk 3 above)
Comment (lannka) [reply src/inabox/host-services.js]: done (after merging with master)

**Comment by lannka** [comment]
Tested in native apps.

**Pull request merged**